pyproject.toml

```diff
@@ -189,6 +191,7 @@ disable = [
   "protected-access",  # caught by ruff
   "raise-missing-from",  # caught by ruff
   "redefined-builtin",  # caught by ruff
+  "singleton-comparison",  # caught by ruff
```

Comment:
Is this actually caught by ruff?

Reply:
Yes, but ignored in ruff as of now.
This is pseudo code from src file.